polaris/ocean/tasks/ice_shelf_2d/forward.py

```diff
@@ -0,0 +1,161 @@
+from polaris.mesh.planar import compute_planar_hex_nx_ny
+from polaris.ocean.model import OceanModelStep, get_time_interval_string
+
+
+class Forward(OceanModelStep):
+    """
+    A step for performing forward ocean component runs as part of baroclinic
+    channel tasks.
+
+    Attributes
+    ----------
+    resolution : float
+        The resolution of the task in km
+    """
+    def __init__(self, component, resolution, mesh, init,
+                 name='forward', subdir=None, indir=None,
+                 ntasks=None, min_tasks=None, openmp_threads=1,
+                 do_restart=False):
+        """
+        Create a new task
+
+        Parameters
+        ----------
+        component : polaris.Component
+            The component the step belongs to
+
+        resolution : km
+            The resolution of the task in km
+
+        name : str
+            the name of the task
+
+        subdir : str, optional
+            the subdirectory for the step.  If neither this nor ``indir``
+             are provided, the directory is the ``name``
+
+        indir : str, optional
+            the directory the step is in, to which ``name`` will be appended
+
+        ntasks : int, optional
+            the number of tasks the step would ideally use.  If fewer tasks
+            are available on the system, the step will run on all available
+            tasks as long as this is not below ``min_tasks``
+
+        min_tasks : int, optional
+            the number of tasks the step requires.  If the system has fewer
+            than this number of tasks, the step will fail
+
+        openmp_threads : int, optional
+            the number of OpenMP threads the step will use
+        """
+        if do_restart:
+            name = 'restart'
+        super().__init__(component=component, name=name, subdir=subdir,
+                         indir=indir, ntasks=ntasks, min_tasks=min_tasks,
+                         openmp_threads=openmp_threads)
+
+        self.resolution = resolution
+        self.do_restart = do_restart
+
+        # make sure output is double precision
+        self.add_yaml_file('polaris.ocean.config', 'output.yaml')
+
+        self.add_input_file(filename='init.nc',
+                            work_dir_target=f'{init.path}/output.nc')
+        self.add_input_file(filename='graph.info',
+                            work_dir_target=f'{mesh.path}/culled_graph.info')
+        if do_restart:
+            self.add_input_file(filename='restarts',
+                                target='../forward/restarts')
```

Comment:
I don't think a directory can be used as an input file.  It might work right now but that won't work once task parallelism is added, since it will actually rely on the presence of absence of the input file to determine if it's safe to run.

Reply:
It might be better to have the restart forward step have the non-restart forward step as a dependency and not worry about the restart files themselves.

Reply:
Let me know if I should provide some example code.

Reply:
I know how to add the step as a dependency. Would you like me to do this whenever there is a step dependency, even when there are input files linked?

Reply:
> I know how to add the step as a dependency. Would you like me to do this whenever there is a step dependency, even when there are input files linked?

No, there's extra overhead involved in adding a step as a dependency so only in cases where you don't know the filename(s) at init.

Reply:
In terms of indicating where to find the restart files, are you content if that is just indicated in the streams file's path, now that I have removed them as input files?